File: src/main/java/cz/dusanrychnovsky/chessengine/core/Column.java
package cz.dusanrychnovsky.chessengine.core;

/**
 * Represents a column on a chessboard - A to H.
 */
public enum Column implements Navigable<Column> {
  CA, CB, CC, CD, CE, CF, CG, CH;

  @Override
  public int getOrd() {
    return ordinal();
  }

  @Override
  public Column[] getAll() {
    return values();
  }

  /**
   * @return Name of the column.
   */
  @Override
  public String toString() {
    return super.toString().substring(1, 2);
  }
}

File: app/src/main/kotlin/cz/dusanrychnovsky/chessengine/Column.kt
package cz.dusanrychnovsky.chessengine

import cz.dusanrychnovsky.chessengine.extensions.next
import cz.dusanrychnovsky.chessengine.extensions.prev

enum class Column : Navigable<Column> {
    CA, CB, CC, CD, CE, CF, CG, CH;

    override fun next(): Column? = this.next<Column>()
    override fun prev(): Column? = this.prev<Column>()
}
